### ISSUE TITLE
Variable has initial value which is impossible by start value ot equations

### ISSUE THREAD

**Issue by paultjevdh** [OpenModelica/OpenModelica#12108]
### Description
During the initialization phase of a model I get a divide by zero warning. However, the equation that sets its value cannot return zero.

### Steps to reproduce
See the enclosed model.
[Malcolm6.zip](https://github.com/OpenModelica/OpenModelica/files/14602255/Malcolm6.zip)

When running in openmodelica 1.22.3 or earlier, the message window reports:
![image](https://github.com/OpenModelica/OpenModelica/assets/59084059/7fbf86b4-9798-4125-9a19-2434e5420bd4)

This error is fixed when a start value is given, but causality dictates that the start value for phi should set Ic unambiguously.
It seems openmodelica ignores this

### Expected behavior
Dymola is able to run the model without complaints, so I expect the model is valid.

### Version and OS
 - OpenModelica Version 1.22.3
 - OS: Windows 11, 64 bit

**Comment by casella** [comment]
BTW, if you check with the debugger, the equation matched with `FASresistance.Ic` is
```
(residual) (if noEvent(abs(FASresistance.i) / FASresistance.Ic <= 1.0) then 0.0 else FASresistance.Rn * sqrt(1.0 - (FASresistance.Ic / abs(FASresistance.i)) ^ 2.0)) - FASresistance.R = 0
```
Probably Dymola can solve it in closed form, while OMC for some reason can't. But that is not really a bug.

**Comment by casella** [comment]
Wait, there is something weird in the causalization of this problem, let me check it further.

**Comment by casella** [comment]
I looked up how the equations are solved in the debugger, and I was seeing strange things, with variables being computed in a weird order. In particular, the way `rowAddress1.y` was initialized was super-weird. Then, I understood what the problem is: _your model lacks initial equations for all continuous states and discrete variables_; in other words, the initial solution is not completely determined by the Modelica code. 

This requires the Modelica tool to figure out where to add some extra `fixed = true` attributes to completely determine the initialization problem. Unfortunately this necessarily requires some heuristics, as there is no agreed-upon algorithm to do that, and in most cases there could be ambiguities. This is super-dangerous, because different tools can use different heuristics, hence end up with different initialization problems, which could turn out to be difficult or impossible to solve, or just plain wrong.

You can see this in the Warnings window:
```
[1] 14:59:14 Translation Warning
Assuming fixed start value for the following 1 variables:
         L_SQ1nyq.i:VARIABLE(start = 1e-6 unit = "A" fixed = true nominal = 1e-6 )  "Current flowing from pin p to pin n"Malcolm, Modelica.Electrical.Analog.Basic.Inductor type: Real
```
OMC found one missing initial equations for discrete variables and decided to fix `ç_SQ1nyq.i`, instead of fixing `rowAddress1.y` to its start value of 1, which I guess was your original intention. From that point onwards, all kind of weird stuff happens.

If you add fixed = true to the extends clause of RowAddress:
```modelica
  extends Modelica.Blocks.Interfaces.SO(y(start=1, fixed = true));
```
the model works just fine in OMC.

Bottom line: if you want to port models from Dymola to OMC (or other Modelica tools) safely, always check the Dymola warnings about missing initial equations, and add `fixed = true` to the variables that Dymola decides to fix in order to obtain an initial solution, if you like the solution that Dymola found. This will guarantee that other tools solve the same initialization problem, and in most cases also ensure that the solution is the same (though this is not guaranteed if the initialization problem is implicit).

**Comment by paultjevdh** [comment]
Thanks for taking this up seriously.

Nevertheless, in this model Ic depends only on parameters and a signal with fixed  'input' causality (FASresistance.phi). 
`FASresistance.Ic = 0.5 * (cos(FASresistance.phi) + 1.00001);`
Therefore _I_ know that there is a causal path to calculate Ic, and the result will be >0, no matter the start value of FASresistance.phi. Openmodelica ignores this, at least during initialization it seems.
If the openmodelica debugger thinks Ic is calculated differently:
`(residual) (if noEvent(abs(FASresistance.i) / FASresistance.Ic <= 1.0) then 0.0 else FASresistance.Rn * sqrt(1.0 - (FASresistance.Ic / abs(FASresistance.i)) ^ 2.0)) - FASresistance.R = 0`,
 it should reconsider its priorities. Causality should overrule start values. 

It worries me that the causality info (input/output) is missing in the flattened model. I hope this is not an intermediate step for the compiler?

PS I was not implying ' nominal' had any function is this model. It was a leftover from reducing the model to something I could post.

**Comment by casella** [comment]
input/output in Modelica has nothing to do with causality. It is only meant to check balancedness. In a connection set you must have one output and one or more inputs.

**Comment by paultjevdh** [comment]
Fritszon calls input and output: causality prefixes, which made me draw some conclusions. Digging a bit further in the specification shows that you are correct, of course.
Thanks for pointing this out.

**Comment by casella** [comment]
> Fritzson calls input and output: causality prefixes
They kind of are. But you can always take a block diagram with 10 cascaded blocks, and then prescribe the output of the last one and get the whole thing computed backwards to get the input (provided the output is smooth enough, of course). Basically getting the inverse system dynamics. All outputs are turned to inputs and vice-versa. 

You really can't fix causality in an equation-based language. The only constraints are about balanced models and not allowing two or more outputs in the same connection set.

**Comment by paultjevdh** [comment]
...so I can set the output value through an initial equation and have modelica determine the corresponding start value for the input... 
I actually use this regularly, so I should have known.

**Comment by casella** [comment]
> ...so I can set the output value through an initial equation

Initialization is one thing. There is no causality with initialization problems, if you initialize in steady-state you can either set the input or the output (unless the linearized system has a pole or a zero in the origin, in which case you can only set the output or the input, respectively). 

What I meant was having a regular equation that sets the values of the output at all times. So you compute the inverse of the model, to determine which input transient is required to get that output transient.

I can post an example of that if you need
